scripts/validate-config

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+"""
+Utility to verify that SecureDrop Workstation config is properly structured.
+Checks for
+"""
+import json
+import re
+import os
+import subprocess
+
+
+TOR_V3_HOSTNAME_REGEX = r'^[a-z2-7]{56}\.onion$'
+TOR_V3_AUTH_REGEX = r'^[A-Z2-7]{52}$'
+
+TOR_V2_HOSTNAME_REGEX = r'^[a-z2-7]{16}\.onion$'
+TOR_V2_AUTH_COOKIE_REGEX = r'^[a-zA-z0-9+/]{22}$'
+
+# CONFIG_FILEPATH = "/srv/salt/sd/config.json"
+CONFIG_FILEPATH = "config.json"
+
+
+class ValidationError(Exception):
+    pass
+
+
+class SDWConfigValidator(object):
+    def __init__(self):
+        self.config_filepath = CONFIG_FILEPATH
+        self.confirm_config_file_exists()
+        self.config = self.read_config_file()
+        self.confirm_onion_config_valid()
+        self.confirm_usb_export_device()
+        self.confirm_submission_privkey_file()
+        self.confirm_submission_privkey_fingerprint()
+
+    def confirm_config_file_exists(self):
+        try:
+            assert os.path.exists(self.config_filepath)
+        except AssertionError:
+            msg = "Config file does not exist: {}".format(self.config_filepath)
+            msg += "Create from config.json.example"
+            raise AssertionError(msg)
+
+    def confirm_onion_config_valid(self):
+        """
+        We support both v2 and v3 Onion Services, so if the values
+        in config file match either format, good enough.
+        """
+        try:
+            self.confirm_onion_v3_url()
+            self.confirm_onion_v3_auth()
+        except AssertionError:
+            self.confirm_onion_v2_url()
+            self.confirm_onion_v2_auth()
+
+    def confirm_onion_v3_url(self):
+        assert "hidserv" in self.config
+        assert "hostname" in self.config["hidserv"]
+        assert re.match(TOR_V3_HOSTNAME_REGEX, self.config["hidserv"]["hostname"])
+
+    def confirm_onion_v3_auth(self):
+        assert "hidserv" in self.config
+        assert "key" in self.config["hidserv"]
+        assert re.match(TOR_V3_AUTH_REGEX, self.config["hidserv"]["key"])
+
+    def confirm_onion_v2_url(self):
+        assert "hidserv" in self.config
+        assert "hostname" in self.config["hidserv"]
+        assert re.match(TOR_V2_HOSTNAME_REGEX, self.config["hidserv"]["hostname"])
+
+    def confirm_onion_v2_auth(self):
+        assert "hidserv" in self.config
+        assert "key" in self.config["hidserv"]
+        assert re.match(TOR_V2_AUTH_COOKIE_REGEX, self.config["hidserv"]["key"])
+
+    def confirm_usb_export_device(self):
+        assert "usb" in self.config
+        assert "device" in self.config["usb"]
+        usb_config = self.config["usb"]["device"]
+        assert usb_config.startswith("sys-usb")
```

Comment:
I realized this time through that this doesn't check for a completely valid Qubes USB identifier; it should technically be another regex check for `r'^sys-usb:\d+-\d+$'`. I'm inclined to let it go at this point, given its impending obsolescence, but could also make that change if anyone feels differently.

Reply:
Yeah, going to merge this as-is and just get the auto attachment in ASAP.